mlir/include/mlir/Interfaces/MemorySlotInterfaces.td

```diff
@@ -298,14 +298,20 @@ def DestructurableAllocationOpInterface
       "destructure",
       (ins "const ::mlir::DestructurableMemorySlot &":$slot,
            "const ::llvm::SmallPtrSetImpl<::mlir::Attribute> &":$usedIndices,
-           "::mlir::OpBuilder &":$builder)
+           "::mlir::OpBuilder &":$builder,
+           "::mlir::SmallVectorImpl<::mlir::DestructurableAllocationOpInterface> &":
+             $newAllocators)
     >,
     InterfaceMethod<[{
         Hook triggered once the destructuring of a slot is complete, meaning the
         original slot is no longer being refered to and could be deleted.
         This will only be called for slots declared by this operation.
+
+        Must return a new destructurable allocation op if this operation
+        produced multiple destructurable slots, nullopt otherwise.
```

Comment:
```suggestion
        Must return a new destructurable allocation op if this hook creates
        new destructurable ops, nullopt otherwise.
```
I assume this is the intended meaning, right? You're not talking about the new allocas of fields, I assume.

Reply:
I'm not sure I understand what you meant actually, could you explain?

Reply:
I was explicitly referring to the case where the alloca originally produced multiple destructurable memory slots. I'll non-the-less update the comment.